src/util/profile/profile.ts

```diff
@@ -83,7 +84,7 @@ class ProfileImpl implements Profile {
     return this;
   }
 
-  setAccessToken(token: TokenValueType): Promise<Profile> {
+  setAccessToken(token: TokenValueType): Promise<void> {
```

Comment:
IIRC, I returned the profile here because we need the returned profile with the updated token in it. Subtle errors occurred otherwise.